www/js/util.js

```diff
@@ -1708,9 +1695,19 @@ function pingMessage(isHighlight) {
             isHighlight)) {
             CHATSOUND.play();
         }
+
+        if (USEROPTS.notifications === "always" || (USEROPTS.notifications === "onlyping" &&
+            isHighlight)) {
+            showDesktopNotification(notificationTitle, notificationBody.replace(/\[[^\]]+\]/g, ''));
+        }
     }
 }
 
+function showDesktopNotification(notificationTitle, notificationBody)
```

Comment:
Not going to lie here, I opted to introduce this function instead of just doing `new Notification()` in `pingMessage` because it will be easier to override in channel JS.